packages/experiments/src/utilities/VariableProcessing.ts

```diff
@@ -1,10 +1,18 @@
-import { merge } from '../Utilities';
+export type IComponentStyleVariables<TComponentViewProps, TComponentStyleVariablesTypes> =
+  | TComponentStyleVariablesTypes
+  | ((props: TComponentViewProps) => TComponentStyleVariablesTypes)
+  | undefined;
 
-export type IProcessedVariables<T> = { [P in keyof T]-?: IProcessedVariables<T[P]> };
+export function resolveStyleVariables<
```

Comment:
minor nit but for utility files, reserve TitleCasing in file names for classes and components. For files which contain a number of exported functions, use camelCase. (e.g. object.ts vs Button.base.tsx